services-camel/src/main/java/edu/unc/lib/dl/services/camel/images/AddDerivativeProcessor.java

```diff
@@ -48,61 +45,41 @@
 public class AddDerivativeProcessor implements Processor {
     private static final Logger log = LoggerFactory.getLogger(AddDerivativeProcessor.class);
 
-    private final RepositoryObjectLoader repoObjLoader;
-    private final String slug;
     private final String fileExtension;
-    private final String mimetype;
-
-    private final int maxRetries;
-    private final long retryDelay;
+    private final String derivativeBasePath;
 
-    public AddDerivativeProcessor(RepositoryObjectLoader repoObjLoader, String slug, String fileExtension,
-            String mimetype, int maxRetries, long retryDelay) {
-        this.repoObjLoader = repoObjLoader;
-        this.slug = slug;
+    public AddDerivativeProcessor(String fileExtension, String derivativeBasePath) {
         this.fileExtension = fileExtension;
-        this.maxRetries = maxRetries;
-        this.retryDelay = retryDelay;
-        this.mimetype = mimetype;
+        this.derivativeBasePath = derivativeBasePath;
     }
 
     @Override
     public void process(Exchange exchange) throws Exception {
         Message in = exchange.getIn();
         String binaryUri = (String) in.getHeader(FCREPO_URI);
-        int retryAttempt = 0;
+        String binarySubPath = (String) in.getHeader(CdrBinarySubPath);
 
         final ExecResult result = (ExecResult) in.getBody();
 
         String derivativePath = new BufferedReader(new InputStreamReader(result.getStdout()))
```

Comment:
Could you make this `derivativeTmpPath`? I seem to be having a hard time following which paths are landing in which variables since everything is derivatives.